library/src/include/utility.h

```diff
@@ -343,6 +343,23 @@ __device__ __host__ inline rocblas_half rocblas_abs(rocblas_half x)
     return t.x;
 }
 
+template <typename>
+struct rocblas_real_type
+{
+};
```

Comment:
should we have default for float and double?

Reply:
I could change it to default to the passed-in type, might be useful in the future. Or could just leave it in case we want to change it in the future since we aren't using it now.

Reply:
Why does this have to be so complicated?

Just define specializations:
```
template <typename>
struct rocblas_real_type
{
};

template <>
struct rocblas_real_type<rocblas_float_complex>
{
   using type = float;
};

template <>
struct rocblas_real_type<rocblas_double_complex>
{
   using type = double;
};

Reply:
> If we think we'll need to use them at some point for T=double/float (which we might) I'll implement them to use their own type.
> I think we should leave the default one empty as it is now.

The only difference to what you described is that I've defined specialisations for float and double. For `hpr`, this isn't needed, but if we refactor say, `scal`, we could use these since we implement both real and complex types.
If we don't want these now, that's fine, we just need to have an agreement so I don't have to keep changing it back and forth for this PR :)

Reply:
What is the definition of 'so complicated' ?